common/src/main/java/whocraft/tardis_refined/common/blockentity/device/AstralManipulatorBlockEntity.java

```diff
@@ -0,0 +1,261 @@
+                    this.clearDisplay();
+                }
+
+                this.clearDisplay();
+            }
+        }
+    }
+
+    public boolean setBlockPos(BlockPos pos, boolean isPointA) {
+
+        if (!this.shouldDisplay) {
+            return false;
+        }
+
+        if (isPointA) {
+            pointABlockPos = pos;
+        } else {
+            pointBBlockPos = pos;
+        }
+
+        if (checkIfDistanceIsTooGreat(pos)) {
+            this.clearDisplay();
+            sendUpdates();
+            return false;
+
+        }
+
+        sendUpdates();
+        return true;
+    }
+
+    private boolean checkIfDistanceIsTooGreat(BlockPos pos) {
+        var distance = pos.distManhattan(getBlockPos());
+        return distance > 25;
+    }
+
+    @Nullable
+    @Override
+    public Packet<ClientGamePacketListener> getUpdatePacket() {
+        return ClientboundBlockEntityDataPacket.create(this);
+    }
+
+    public void sendUpdates() {
+        level.sendBlockUpdated(this.getBlockPos(), level.getBlockState(this.getBlockPos()), level.getBlockState(this.getBlockPos()), Block.UPDATE_CLIENTS);
+        setChanged();
+    }
+
+    public CompoundTag getUpdateTag() {
+        CompoundTag tag = new CompoundTag();
+        saveAdditional(tag);
+        return tag;
+    }
+
+
+    private void attemptToBuild(BlockPos pointA, BlockPos pointB) {
+
+        var submittedBlocks = new ArrayList<ManipulatorCraftingRecipeItem>();
+
+        // Grab the absolutes for these items.
+        var xDiff = Math.abs(pointA.getX() - pointB.getX());
+        var yDiff = Math.abs(pointA.getY() - pointB.getY());
+        var zDiff = Math.abs(pointA.getZ() - pointB.getZ());
+
+        var smallestPointX = pointA.getX() > pointB.getX() ? pointB.getX() : pointA.getX();
+        var smallestPointY = pointA.getY() > pointB.getY() ? pointB.getY() : pointA.getY();
+        var smallestPointZ = pointA.getZ() > pointB.getZ() ? pointB.getZ() : pointA.getZ();
+
+        var minPoint = new BlockPos(smallestPointX, smallestPointY, smallestPointZ);
+
+        for (int y = 0; y < yDiff + 1; y++) {
+
+            for (int x = 0; x < xDiff + 1; x++) {
+
+                for (int z = 0; z < zDiff + 1; z++) {
+
+                    var blockPosInWorld = new BlockPos(minPoint.getX() + x, minPoint.getY() + y, minPoint.getZ() + z);
+
+                    submittedBlocks.add(new ManipulatorCraftingRecipeItem(new BlockPos(x, y, z), level.getBlockState(blockPosInWorld).getBlock()));
+                }
+            }
+        }
+
+        // Filter recipes by the first block, will make it a LOT easier later down the line. (when I make it)
+        var firstBlock = submittedBlocks.stream().filter(x -> x.relativeBlockPos.getX() == 0 && x.relativeBlockPos.getY() == 0 && x.relativeBlockPos.getZ() == 0).findFirst();
+        if (!firstBlock.isPresent()) {
+            return;
+        }
+
+        var possibleRecipes = new ArrayList<ManipulatorCraftingRecipe>();
```

Comment:
var is cool, but let's not use it in Minecraft

Reply:
Is there a documented reason why that's the case?